org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/resource/BinaryResponseHelper.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2010, 2025 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.scout.rt.rest.resource;
+
+import java.io.InputStream;
+
+import jakarta.ws.rs.core.CacheControl;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.Response.ResponseBuilder;
+
+import org.eclipse.scout.rt.platform.ApplicationScoped;
+import org.eclipse.scout.rt.platform.BEANS;
+import org.eclipse.scout.rt.platform.util.DownloadResponseHelper;
+import org.eclipse.scout.rt.platform.util.ObjectUtility;
+
+/**
+ * Helper class used by REST resources.
+ */
+@ApplicationScoped
+public class BinaryResponseHelper {
+
+  public static final String IMAGE_TYPES = "image/jpeg, image/gif, image/png";
+  public static final String ZIP_TYPE = "application/zip";
+
+  /**
+   * Creates a response with streaming binary data and no-cache headers. Use this method to return large data / files.
+   */
+  public Response binaryResponse(InputStream in, String filename, String mimeType) {
+    return binaryResponse(Response.ok(in, mimeType), filename, null);
+  }
+
+  public Response binaryResponse(ResponseBuilder builder, String filename) {
```

Comment:
add javadoc to method

In difference to binaryResponse(byte[] data, String mimeType) method, this method does not add a no-cache-header? On purpose? Mention in javadoc.